### PR TITLE
T11: Digital Root

### PR DESCRIPTION
**Pull request by Dakshita-Minocha**

### CODE REVIEW

**Review by v-priya — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+         for (; ; ) {
+            Console.Write ("Enter number or press [Q] to exit: ");
+            string input = Console.ReadLine ();
+            if (input.ToLower () == "q") Environment.Exit (0);
+            int.TryParse (input, out int num);
+            int i = input.Length, sum;
+            if (num <= 0)
+               Console.WriteLine ("\nError: Digital root is only defined for natural numbers. Try again.\n");
+            else {
+               while (i > 1) {
```
[the remark this answers is not in the document]
Comment (Dakshita-Minocha) [reply Program.cs:15]: okay this works better for single digits also, thanks!

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,34 @@
...
+      /// <summary> Program to find digital root of a number. 
+      /// Digital root = sum of digits calculated iteratively till there's a single digit left.</summary>
+      static void Main () {
+         string input;
+         for (; ; ) {
+            for (; ; ) {
+               Console.Write ("Enter number or press [Q] to exit: ");
+               input = Console.ReadLine ();
+               if (input.ToLower () == "q") Environment.Exit (0);
+               if (input.All (c => Char.IsDigit (c) | c == '-')) break;
```
Comment (v-priya) [inline Program.cs:12]: This 'or' condition will accept '-' char as input.

**Review by v-priya — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 2 above)
Comment (Dakshita-Minocha) [reply Program.cs:12]: Yes I added that because I wanted it to accept negative numbers as well.

Comment (v-priya) [reply Program.cs:12]: But, after accepting the input you are checking for numbers <= 0 which is contradictory.
```
if (num <= 0)
   Console.WriteLine ("\nError: Digital root is only defined for natural numbers. Try again.\n");
```

**Review by v-priya — APPROVED** [review]

**Pull request closed without merging**